### PR TITLE
Add appsub status reference to subcription CRD

### PR DESCRIPTION
**Pull request by philipwu08**
https://github.com/stolostron/backlog/issues/20331

Signed-off-by: Philip Wu <phwu@redhat.com>

### CODE REVIEW

**Comment by mikeshng** [comment]
It might be missing code gen changes? Same comment here https://github.com/open-cluster-management-io/multicloud-operators-subscription/pull/133#issuecomment-1074381152

File: pkg/controller/subscription/subscription_controller.go [hunk 1]
```diff
@@ -250,6 +251,16 @@ func (r *ReconcileSubscription) Reconcile(ctx context.Context, request reconcile
 				klog.Errorf("doReconcile got error %v", reconcileErr)
 			}
 
+			// Update AppstatusReference
+			if !r.standalone {
```
Comment (xiangjingli) [inline pkg/controller/subscription/subscription_controller.go:255]: Does it mean the AppStatus reference is only visible for non standalone appsub?  It looks not necessary.

For standalone appsub, It looks the AppStatusreference is still valid. I just checked some standalone appsub e.g
```
% oc get appsub -n open-cluster-management          cluster-lifecycle-sub          
NAME                    STATUS       AGE     LOCAL PLACEMENT   TIME WINDOW
cluster-lifecycle-sub   Subscribed   5h11m   true      
```
And its appsubstatus is 
```
% oc get appsubstatus -n open-cluster-management          cluster-lifecycle-sub 
NAME                    AGE
cluster-lifecycle-sub   5h12m
```

**Review by xiangjingli — COMMENTED** [review]
@philipwu08  BTW, after the PR is merged, I will create a new community CSV to bundle the new appsub CRD.  I will let you know when the new community CSV is merged. You can verify this using a new ACM upstream build

File: pkg/controller/subscription/subscription_controller.go (hunk 1 above)
Comment (philipwu08) [reply pkg/controller/subscription/subscription_controller.go:255]: XJ, you're right with the standalone case. Let me update that.

**Pull request merged**